synapse/module_api/__init__.py

```diff
@@ -573,6 +574,23 @@ def get_qualified_user_id(self, username: str) -> str:
             return username
         return UserID(username, self._hs.hostname).to_string()
 
+    def normalize_username(
+        self, username: Union[str, bytes], case_sensitive: bool = False
+    ) -> str:
+        """Map a username onto a string suitable for a MXID
```

Comment:
Maybe add a hyperlink to the spec here? It's not just that we've got a reasonable-looking mapping, but that it's the specific one defined in the spec.

Reply:
Good idea, sorted.